### PR TITLE
removed pip imports

### LINKED ISSUE #1682: Remove pip imports in favor of importlib_metadata
See https://github.com/hyperledger/indy-plenum/pull/1548 for details.

Make the same changes to indy-node to remove the dependency on pip.

### PR DESCRIPTION
**Pull request by pSchlarb**
Fixes #1682
Signed-off-by: Philipp Schlarb <p.schlarb@esatus.com>

### CODE REVIEW

**Comment by WadeBarnes** [comment]
(ci) test this please

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
There are still occurrences of a pinned version of pip being installed that should be removed.  Occurrences of:
- `pip<10.0.0`
- `pip==10.0.0`

**Comment by pSchlarb** [comment]
Pip dropped support for python3.5 with version 21.
The updates of pip in the base image would upgrade pip beyond that. 
Best would be to stick with the pip version installed by apt.
Other occurences of pinned pip versions removed.

The changes to .github/workflows/build/Dockerfile and environment/docker/baseimage/Makefile except the version change don't need to be merged.
They were done to use the new baseimages for testing.

File: .github/workflows/build/Dockerfile
Comment (WadeBarnes) [inline .github/workflows/build/Dockerfile:1]: You'll need to remove this change from your PR.

File: environment/docker/baseimage/Makefile
Comment (WadeBarnes) [inline environment/docker/baseimage/Makefile:4]: You'll need to remove this change from your PR.

File: environment/docker/baseimage/indy-baseci.ubuntu.dockerfile
Comment (WadeBarnes) [inline environment/docker/baseimage/indy-baseci.ubuntu.dockerfile:1]: You'll need to remove this from your PR.

File: environment/docker/baseimage/indy-baseci.version
Comment (WadeBarnes) [inline environment/docker/baseimage/indy-baseci.version:1]: You'll need to remove this from your PR.

File: environment/docker/baseimage/indy-baseimage.version
Comment (WadeBarnes) [inline environment/docker/baseimage/indy-baseimage.version:1]: You'll need to remove this from your PR.

File: environment/docker/baseimage/indy-core-baseci.ubuntu.dockerfile
Comment (WadeBarnes) [inline environment/docker/baseimage/indy-core-baseci.ubuntu.dockerfile:1]: You'll need to remove this from your PR.

File: environment/docker/baseimage/indy-core-baseci.version
Comment (WadeBarnes) [inline environment/docker/baseimage/indy-core-baseci.version:1]: You'll need to remove this from your PR.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]

**Comment by pSchlarb** [comment]
Wouldn't the requested changes break the version history of the Docker Basefiles?
According to  https://github.com/hyperledger/indy-node/blob/master/environment/docker/baseimage/README.md the changes in the Dockerfiles lead to updated version files and according tags in the FROM Statement of the Dockerfiles.

Also since the Github Workflow Container relies on the baseimage, it seems that the whole pipeline would fail if the baseimage is not updated first.

**Comment by WadeBarnes** [comment]
You are correct.  My concern was the fact that the code is now referencing your Docker Hub repository; `pschlarb`.  That will need to change, and we'll need to prepare for a new release of the base images to the `hyperledger` Docker Hub repository.

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by pSchlarb** [comment]
> The changes to .github/workflows/build/Dockerfile and environment/docker/baseimage/Makefile except the version change don't need to be merged.

That was exactly what i meant with that. Sorry for being unclear.

Do I need to change the DockerHub Repository, or is that done while merging?

**Comment by WadeBarnes** [comment]
It will need to be done before merging, but hold off until we have all the tests passing first.

File: setup.py
Comment (udosson) [inline setup.py:30]: Can be removed here because `importlib-metadata` is part of `install-requires`

File: setup.py
Comment (udosson) [inline setup.py:58]: Do we need to pin this to version `1.7.0`?
In plenum we use `importlib_metadata>=2.0` 
https://github.com/hyperledger/indy-plenum/blob/406afdeca1630be688f803a3cba15115faa20e2b/setup.py#L111

File: acceptance/indy-cli-batches/AS-03-01-identity-owner-anyone-can-write.batch
Comment (pSchlarb) [reply acceptance/indy-cli-batches/AS-03-01-identity-owner-anyone-can-write.batch:1]: On a clean clone of the repo: 
```
root@73570c5a4ed8:/indy-node# git diff -w
warning: CRLF will be replaced by LF in acceptance/indy-cli-batches/AS-03-01-identity-owner-anyone-can-write.batch.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in acceptance/indy-cli-batches/AS-03-01-steward.batch.
The file will have its original line endings in your working directory.
```

Due to 

> # Set the default behavior, in case people don't have core.autocrlf set.
> * text=auto

in .gitattributes.

If LF is the desired line ending, these files have to change either in this PR or another one.

File: acceptance/indy-cli-batches/AS-03-01-steward.batch
Comment (pSchlarb) [reply acceptance/indy-cli-batches/AS-03-01-steward.batch:1]: On a clean clone of the repo: 
```
root@73570c5a4ed8:/indy-node# git diff -w
warning: CRLF will be replaced by LF in acceptance/indy-cli-batches/AS-03-01-identity-owner-anyone-can-write.batch.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in acceptance/indy-cli-batches/AS-03-01-steward.batch.
The file will have its original line endings in your working directory.
```

Due to 

> # Set the default behavior, in case people don't have core.autocrlf set.
> * text=auto

in .gitattributes.

If LF is the desired line ending, these files have to change either in this PR or another one.

**Comment by pSchlarb** [comment]
The needs for changing the Baseimages are:
- Pip was pinned to <=10
- Pinning of setuptools<=50.3.2 since it dropped support for python3.5
- Pinning of importlib_metadata <3.0 since it dropped support for python3.5 (it would install 4.8.1 with >=2.0)

The test would fail, because of the incompatible importlib_metadata version.

File: ci/pipeline.groovy
Comment (udosson) [inline ci/pipeline.groovy:151]: `pip install 'pyzmq==18.1.0'` is required

**Comment by udosson** [comment]
@pSchlarb could you please remove all the changes you made in `./environment/docker`, please?

**Comment by udosson** [comment]
(ci) test this please

File: Jenkinsfile.ci
Comment (udosson) [inline Jenkinsfile.ci:76]: `sh "pip install 'pyzmq==18.1.0'"` is required here as well

File: setup.py
Comment (udosson) [inline setup.py:59]: Change to `indy-plenum==1.13.0.dev169`

**Review by udosson — CHANGES_REQUESTED** [review]
@pSchlarb, the PR is almost done. Thanks in advance for adjusting once more! :)

File: setup.py
Comment (pSchlarb) [reply setup.py]: no problem :D

**Comment by udosson** [comment]
(ci) test this please

File: Jenkinsfile.ci
Comment (udosson) [reply Jenkinsfile.ci:76]: @pSchlarb

File: setup.py
Comment (pSchlarb) [reply setup.py]: Strangely the test collection will fail with the newer version of plenum.

```
_____________________________________________________________________ ERROR collecting indy_node/test/auth_rule/auth_framework/test_auth_rule_using.py _____________________________________________________________________
ImportError while importing test module '/indy-node/indy_node/test/auth_rule/auth_framework/test_auth_rule_using.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
indy_node/test/auth_rule/auth_framework/test_auth_rule_using.py:7: in <module>
    from indy_node.test.auth_rule.auth_framework.edit_frozen_ledgers import EditFrozenLedgersTest
indy_node/test/auth_rule/auth_framework/edit_frozen_ledgers.py:10: in <module>
    from plenum.test.freeze_ledgers.helper import sdk_send_freeze_ledgers
/usr/local/lib/python3.5/dist-packages/plenum/test/freeze_ledgers/helper.py:3: in <module>
    from indy.ledger import build_ledgers_freeze_request, build_get_frozen_ledgers_request
E   ImportError: cannot import name 'build_ledgers_freeze_request'
________________________________________________________________________ ERROR collecting indy_node/test/freeze_ledgers/test_send_ledgers_freeze.py ________________________________________________________________________
ImportError while importing test module '/indy-node/indy_node/test/freeze_ledgers/test_send_ledgers_freeze.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
indy_node/test/freeze_ledgers/test_send_ledgers_freeze.py:4: in <module>
    from plenum.test.freeze_ledgers.helper import sdk_get_frozen_ledgers, sdk_send_freeze_ledgers
/usr/local/lib/python3.5/dist-packages/plenum/test/freeze_ledgers/helper.py:3: in <module>
    from indy.ledger import build_ledgers_freeze_request, build_get_frozen_ledgers_request
E   ImportError: cannot import name 'build_ledgers_freeze_request'
```
In the [run](https://github.com/hyperledger/indy-node/actions/runs/1504681011) without the new plenum versions the test were running.

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by WadeBarnes** [comment]
@udosson, All the tests are passing now.  Do you want to give this one last look?

File: setup.py
Comment (WadeBarnes) [reply setup.py]: This is a sign that the version of the `indy-sdk` being installed does not contain `indy.ledger.build_ledgers_freeze_request`, but the version of node contains the `freeze_ledgers` tests that require it.

Comment (WadeBarnes) [reply setup.py]: This should be resolved when this PR is merged; https://github.com/hyperledger/indy-plenum/pull/1578

**Comment by WadeBarnes** [comment]
@udosson, scratch that, we should wait for the PR and then update to the new version of `indy-plenum`; https://github.com/hyperledger/indy-node/pull/1684#discussion_r777040359

**Comment by WadeBarnes** [comment]
@pSchlarb, [indy-plenum_1.13.0~dev178](https://hyperledger.jfrog.io/ui/repos/tree/General/indy/pool/xenial/main/i/indy-plenum/indy-plenum_1.13.0~dev178_amd64.deb) has been published.  Please try updating to `indy-plenum==1.13.0.dev178`

File: setup.py
Comment (WadeBarnes) [inline setup.py:32]:
```suggestion
                 'python3-indy==1.16.0-dev-1636', 'pytest-asyncio==0.8.0']
```

File: setup.py
Comment (WadeBarnes) [inline setup.py:59]:
```suggestion
    install_requires=['indy-plenum==1.13.0.dev178',
```

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
@pSchlarb, Here are the changes that need to be made.

**Comment by udosson** [comment]
@pSchlarb and @WadeBarnes I'll review the PR once the plenum package has been updated.

**Comment by WadeBarnes** [comment]
(ci) test this please

**Review by WadeBarnes — APPROVED** [review]

**Review by udosson — APPROVED** [review]
LGTM. Thanks @pSchlarb

**Pull request merged**
